adapters/repos/db/vector/hnsw/search_with_max_dist.go

```diff
@@ -19,7 +19,7 @@ import (
 )
 
 func (h *hnsw) KnnSearchByVectorMaxDist(searchVec []float32, dist float32,
-	ef int, allowList helpers.AllowList) ([]int, error) {
+	ef int, allowList helpers.AllowList) ([]int64, error) {
```

Comment:
whole file: see above, all node ids should be represented by `uint64`, removing the need for the conversions.